daemon/backends/macos_proxy_controller.py

```diff
@@ -0,0 +1,42 @@
+import subprocess
+from daemon.backends._proxy_controller import ProxyController
+
+
+class macOSProxyController(ProxyController):
+    """Lento proxy controller using `networksetup` on macOS."""
+
+    def __init__(self):
+        super().__init__()
+
+    def softblock_prompt(self, site):
```

Comment:
Of note: the softblock prompts are platform specific, on macOS I'm running an AppleScript command, on Windows I'm using a built-in DLL. This is because PySide doesn't let you run GUIs on child threads (at least on macOS) and Tkinter was not cooperating. I'll probably come back here at some point and make the softblock prompt use PySide through some sort of signal/slot-based system to reconnect with the main thread but that's an issue to figure out later.